cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/RepositoryManagerTest.java

```diff
@@ -144,14 +147,188 @@ public void testValidateNullBranchName() {
-    RepositoryManager.validateConfig(secureStore, sourceControlConfig);
+    RepositoryManager.validateConfig(secureStore, getSourceControlConfig());
+  }
+
+  @Test
+  public void testGetFileHashSuccess() throws Exception {
+    String fileName = "pipeline.json";
+    Path path = Paths.get(fileName);
+    addFileToGit(path, "{name: pipeline}");
+    RepositoryManager manager = getRepositoryManager();
+    manager.startUp();
+    String fileHash = manager.getFileHash(manager.getBasePath().resolve(fileName));
+    // Excepted hash is found by creating a file with the same name and contents and running "git ls-files -s
+    // <file-name>"
+    Assert.assertEquals(fileHash, "5905258bb958ceda80b6a37938050ad876920f09");
+    // Change the file, but don't commit. The hash should remain the same.
+    Files.write(manager.getBasePath().resolve(fileName), "change 1".getBytes(StandardCharsets.UTF_8));
+    fileHash = manager.getFileHash(manager.getBasePath().resolve(fileName));
+    Assert.assertEquals(fileHash, "5905258bb958ceda80b6a37938050ad876920f09");
+    manager.shutDown();
+
+    // Change the file contents and commit, the hash should also change.
+    addFileToGit(path, "change 2");
+    manager.startUp();
+    fileHash = manager.getFileHash(manager.getBasePath().resolve(fileName));
+    Assert.assertNotEquals(fileHash, "5905258bb958ceda80b6a37938050ad876920f09");
+    manager.shutDown();
+  }
+
+  @Test
+  public void testGetFileHashUncommittedFile() throws Exception {
+    String fileName = "pipeline.json";
+    RepositoryManager manager = getRepositoryManager();
+    manager.startUp();
+    // Change the file, but don't commit.
+    Files.write(manager.getBasePath().resolve(fileName), "change 1".getBytes(StandardCharsets.UTF_8));
+    boolean exceptionCaught = false;
+    try {
+      manager.getFileHash(manager.getBasePath().resolve(fileName));
+    } catch (NotFoundException e) {
+      exceptionCaught = true;
+    }
+    Assert.assertTrue(exceptionCaught);
+    manager.shutDown();
+  }
+
+  @Test
+  public void testGetFileHashDirectory() throws Exception {
+    String fileName = "pipeline.json";
+    String dirName = "data-pipelines";
+    // Add a regular file inside a directory.
+    Path regularFilePath = Paths.get(dirName, fileName);
+    addFileToGit(regularFilePath, "{name: pipeline}");
+    // Add a symbolic link.
+    String symbolicLinkFileName1 = "pipeline-link-1.json";
+    String symbolicLinkFileName2 = "pipeline-link-2.json";
+    Path symlinkPath1 = Paths.get(symbolicLinkFileName1);
+    addSymbolicLinkToGit(symlinkPath1, regularFilePath);
+    // Create a symbolic link to a symbolic link (a->b->c).
+    addSymbolicLinkToGit(Paths.get(symbolicLinkFileName2), symlinkPath1);
+    RepositoryManager manager = getRepositoryManager();
+    manager.startUp();
+    boolean exceptionsCaught = false;
+    try {
+      manager.getFileHash(manager.getBasePath().resolve(dirName));
+    } catch (IllegalArgumentException e) {
```

Comment:
Please use `assertthrows` or `@Test(expected = <Exception>.class)`
IMHO we do not need to test the message